client/my-sites/themes/themes-selection.jsx

```diff
@@ -161,16 +159,7 @@ const ConnectedThemesSelection = connect(
 			isRequesting: isRequestingThemesForQuery( state, siteIdOrWpcom, query ),
 			isLastPage: isThemesLastPageForQuery( state, siteIdOrWpcom, query ),
 			isThemeActive: themeId => isThemeActive( state, themeId, siteId ),
-			isThemePurchased: themeId => (
-				// Note: This component assumes that purchase and data is already present in the state tree
-				// (used by the isThemePurchased selector). At the time of implementation there's no caching
-				// in <QuerySitePurchases /> and a parent component is already rendering it. So to avoid
-				// redundant AJAX requests, we're not rendering the query component locally.
-				isThemePurchased( state, themeId, siteId ) ||
-				// The same is true for the `hasFeature` selector, which relies on the presence of
-				// a `<QuerySitePlans />` component in a parent component.
-				hasFeature( state, siteId, FEATURE_UNLIMITED_PREMIUM_THEMES )
-			),
```

Comment:
As discussed on Slack, we should keep the essence of this comment (tho we need to rephrase it a bit since it's only one selector now). Maybe something like

```js
// Note: This component assumes that purchase and plans data is already present in the state tree
// (used by the `isPremiumThemeAvailable` selector). That data is provided by the `<QuerySitePurchases />`
// and `<QuerySitePlans />` components, respectively. At the time of implementation, neither of them
// provides caching, and both are already being rendered by a parent component. So to avoid
// redundant AJAX requests, we're not rendering these query components locally.
```